File: test/Github-Analyzer.Analysis.Tests/Fixtures/CSharp/Services/UserService.cs
using GithubAnalyzer.Fixtures.Models;

namespace GithubAnalyzer.Fixtures.Services;

public class UserService
{
    private readonly List<User> _users = new();

    public UserService()
    {
    }

    public User FindById(int id)
    {
        return _users.FirstOrDefault(u => u.Id == id)
            ?? throw new InvalidOperationException("User not found.");
    }

    public List<User> GetAll()
    {
        return _users.ToList();
    }

    public void Save(User user)
    {
        _users.Add(user);
    }
}

File: test/Github-Analyzer.Analysis.Tests/Fixtures/Php/Services/UserService.php
<?php

namespace App\Services;

class UserService
{
    private array $users = [];

    public function __construct()
    {
    }

    public function findById(int $id): array
    {
        return $this->users[$id] ?? [];
    }

    public function getAll(): array
    {
        return $this->users;
    }

    public function save(string $name, string $email): void
    {
        $this->users[] = ['name' => $name, 'email' => $email];
    }
}
